### ISSUE TITLE
Everybody can "fulfil" my wishes

### ISSUE THREAD

**Issue by foresle** [wishthis/wishthis#90]
I found problem with UX in your app. Let's image the next situation, I created my list and add all my wishes (20), then I shared this list to my friends or everyone. Some dude cans "fulfil" all my wishes without any authorization, so I will lose all my wishes, but I can't restore it or just see who made this.

I think if you restrict fulfilling wishes without auth and made notifications about fulfilling by another user, it can solve the problem.

Also, I think will be nice to create a request system when owner of wishes can deny or approve fulfilling his wishes by another user.

**Comment by grandeljay** [comment]
> Some dude cans "fulfil" all my wishes without any authorization,

Indeed. This was actually intended since I don't want to require people _having_ to create an account just so they can fulfil wishes. I have a lot of people in my circle which are very unfamiliar with a computer and so I wanted to make the fulfilment process as easy as possible with as little road blocks possible.

That's why the shared wishlist URLs are so hard to guess (virtually impossible). Be mindful who you send it to.

> so I will lose all my wishes, but I can't restore it or just see who made this.

That must indeed be frustrating. The wishes are actually not supposed to disappear for you, just for everybody else. This is unfortunately a severe bug in the current release, sorry about that.

> Also, I think will be nice to create a request system when owner of wishes can deny or approve fulfilling his wishes by another user.

That's an interesting thought and I will consider it in the further development process!
